test/express.ts

```diff
@@ -181,6 +182,63 @@ void tap.test(
 	}
 )
 
+void tap.test(
+	'maintains unique value when the library is depended upon both directly and transitively',
+	async (t) => {
+		const app = express()
+
+		// this function in the test library does app.use(middleware) and
+		// httpContext.set(REQUEST_ID_CONTEXT_KEY, <a unique id>)
+		init(app)
+
+		app.get('/', ((req, res) => {
+			set('value', req.query['value'])
+
+			res.status(200).json({
+				value: req.query['value'],
+				valueFromContext: get<number>('value'),
+				requestId: get<string>(REQUEST_ID_CONTEXT_KEY)
```

Comment:
Please see the code in https://github.com/oliverlockwood/express-http-context-intermediate-library/blob/main/src/index.ts.

We're trying to do two things here:
1. Making use of the fact that the library already called `app.use(middleware)` - so we **should** be able to do our inline get/set of the context `value`
2. Making use of the fact that the library already stored a `requestId` into the context - so we **should** be able to extract it here.

As per the commented assertions below, both usages are failing because of the bug flagged in #1.